### ISSUE TITLE
CommandParameter is null when assigned inside a DataTemplate

### ISSUE THREAD

**Issue by okipol88** [dotnet/wpf#7131]
* .NET Core Version: 6.0 ; 5.0
* Windows version: 21H2 (OS Build 19044.2006)
* Does the bug reproduce also in WPF for .NET Framework 4.8?: Yes

 
 
When using a Data Template and specyfying a command parameter (before the command) the command gets a null value.

Having:
``` xml       
<ResourceDictionary>
  <sys:Boolean xmlns:sys="clr-namespace:System;assembly=mscorlib" x:Key="false">False</sys:Boolean>
</ResourceDictionary>
```

Consider this relay command:

``` cs
 public class BooleanCommand : MarkupExtension, ICommand
  {
    public event EventHandler CanExecuteChanged;

    public bool CanExecute(object parameter)
    {
      if (parameter is Boolean b)
      {
        return true;
      }

      return false;
    }

    public void Execute(object parameter)
    {
    }

    public override object ProvideValue(IServiceProvider serviceProvider)
      => this;
  }
```

 **Actual behavior:** 
The command inside a data template gets a null instead of the defined command parameter boolean value.
 
 **Expected behavior:**
The boolean value should be passed as the command parameter.
 
 **Minimal repro:**

Minimal example [here](https://github.com/okipol88/CommandParameterNullInDataTemplate).

This does not work:
``` xml
<ContentPresenter>
  <ContentPresenter.ContentTemplate>
    <DataTemplate>
        <Button Content="Data Template Button" CommandParameter="{StaticResource false}"  Command="{local:BooleanCommand}" />
    </DataTemplate>
  </ContentPresenter.ContentTemplate>
</ContentPresenter>
```

This however works:
``` xml
<Grid>
  <Button Content="Data Template Button" CommandParameter="{StaticResource false}"  Command="{local:BooleanCommand}" />
</Grid>
```

**Comment by miloush** [comment]
See #4472 and all the duplicates mentioned there. A PR fixing it has been already merged #4217.

**Comment by okipol88** [comment]
> See #4472 and all the duplicates mentioned there. A PR fixing it has been already merged #4217.

Is this really the issue here? The Command parameter is already set and is before the command entry.
That being said the same code:

``` cs
<Button Content="Data Template Button" CommandParameter="{StaticResource false}"  Command="{local:BooleanCommand}" />
```

Behaves differently when used as is inside a `UserControl` and differently when used inside a DataTemplate.

I had a look at the https://github.com/dotnet/wpf/commit/7dd94fe3000a05e93eefeb7cfa68124ad2c32f4f merge so this will fix it by responding to the re-assgnment of the command (hope it behaves the same in the DataTemplate).

**Comment by okipol88** [comment]
I checked it and the change fixes the issue. Marking as closed, applying workaround and will take the official fix in the future :D.
